route/build/cpl/RtmMod.F90

```diff
@@ -711,30 +721,23 @@ SUBROUTINE get_river_export_data(NETOPO_in, RCHFLX_in)
 
         implicit none
         ! Arguments:
-        type(RCHTOPO), intent(in) :: NETOPO_in(:)
-        type(STRFLX),  intent(in) :: RCHFLX_in(:,:)
+        type(RCHTOPO), intent(in)     :: NETOPO_in(:)
+        type(STRFLX),  intent(in)     :: RCHFLX_in(:,:)
+        integer, optional, intent(in) :: offset
         ! Local variables:
         integer                   :: ix
         integer                   :: iens=1
         integer                   :: iRch, iHru
         integer                   :: nRch, nCatch
-        real(r8)                  :: vol_hru
-
-        ! reset volr to zero
-        rtmCTL%volr(:) = 0._r8
 
         nRch=size(NETOPO_in)
         do iRch =1, nRch
           nCatch = size(NETOPO_in(iRch)%HRUIX)
           do iHru = 1, nCatch
             ix = NETOPO_in(iRch)%HRUIX(iHru)
-
+            if (present(offset)) ix=ix+offset
             ! stream volume is split into HRUs based on HRU's areal weight for contributory area
-            vol_hru = RCHFLX_in(iens,iRch)%ROUTE(idxIRF)%REACH_VOL(1) * NETOPO_in(iRch)%HRUWGT(iHru)
-            if (vol_hru > 0._r8) then
-              rtmCTL%volr(ix) = vol_hru/rtmCTL%area(ix)
-            end if
-
+            rtmCTL%volr(ix) = RCHFLX_in(iens,iRch)%ROUTE(idxIRF)%REACH_VOL(1)*NETOPO_in(iRch)%HRUWGT(iHru)/rtmCTL%area(ix)
```

Comment:
The difference to above is that if vol_hru is negative it won't be set to zero. So is vol_hru always zero or positive? Or is it OK to allow it to be negative?

Reply:
Yes, volr should be always 0 or positive (REACH_VOL is always >=0  HRUWGT>0, rtmCTL%area>0).

Reply:
Check the values in NETOPO in check_river_properties in process_ntopo.f90